src/command.rs

```diff
@@ -0,0 +1,141 @@
+use crate::builtins::builtin;
+use crate::builtins::is_builtin;
+use crate::traits::{Runnable, ShellCommand};
+use nix::unistd::{dup2, fork, pipe, ForkResult};
+use std::error::Error;
+use std::os::fd::AsRawFd;
+use std::process::{ChildStdout, Command, Stdio};
+
+pub fn cmd(tokens: Vec<String>) -> Result<Box<dyn ShellCommand>, Box<dyn Error>> {
+    if tokens.is_empty() {
+        Err("Tokens cannot be empty".into())
+    } else if is_builtin(&tokens[0]) {
+        Ok(Box::new(BuiltinCommand::new(tokens)?))
+    } else {
+        Ok(Box::new(ExternalCommand::new(tokens)?))
+    }
+}
+
+pub fn runnable(tokens: Vec<String>) -> Result<Box<dyn Runnable>, Box<dyn Error>> {
+    if tokens.is_empty() {
+        Err("Tokens cannot be empty".into())
+    } else if is_builtin(&tokens[0]) {
+        Ok(Box::new(BuiltinCommand::new(tokens)?))
+    } else {
+        Ok(Box::new(ExternalCommand::new(tokens)?))
+    }
+}
+
+#[derive(Debug)]
```

Comment:
Out of interest what does this do?

Reply:
Let's you print the struct in a debug statement I think.

```rust
dbg!("{:?}")
```